sources/api/netdog/src/net_config/devices/bonding.rs

```diff
@@ -0,0 +1,184 @@
+        if let Some(config) = &self.static4 {
+            config.validate()?
+        }
+        Ok(())
+    }
+
+    fn validate_static6(&self) -> Result<()> {
+        if let Some(config) = &self.static6 {
+            config.validate()?
+        }
+        Ok(())
+    }
+}
+
+impl HasDhcp for &NetBondV1 {
+    fn has_dhcp(&self) -> bool {
+        self.dhcp4.is_some() || self.dhcp6.is_some()
+    }
+}
+
+impl HasRoutes for &NetBondV1 {
+    fn has_routes(&self) -> bool {
+        self.routes.is_some()
+    }
+}
+
+impl Validate for NetBondV1 {
+    fn validate(&self) -> Result<()> {
+        validate_addressing(self)?;
+
+        let interfaces_count: u32 = self.interfaces.len() as u32;
+        ensure!(
+            interfaces_count > 0,
+            error::InvalidNetConfigSnafu {
+                reason: "bonds must have 1 or more interfaces specified"
+            }
+        );
+        if let Some(min_links) = self.min_links {
+            ensure!(
+                min_links <= interfaces_count,
+                error::InvalidNetConfigSnafu {
+                    reason: "min-links is greater than number of interfaces configured"
+                }
+            )
+        }
+        // Validate anything in monitoring configs
+        match &self.monitoring_config {
+            BondMonitoringConfig::MiiMon(config) => {
+                ensure!(
+                    config.frequency > 0,
+                    error::InvalidNetConfigSnafu {
+                        reason: "miimon-frequency of 0 disables Mii Monitoring, either set a value or configure Arp Monitoring"
+                    }
+                );
+                // updelay and downdelay should be a multiple of frequency, but will be rounded down
+                // by the kernel, this ensures they are at least the size of frequency (non-zero)
+                ensure!(
+                    config.frequency <= config.updelay && config.frequency <= config.downdelay,
+                    error::InvalidNetConfigSnafu {
+                        reason: "miimon-updelay and miimon-downdelay must be equal to or larger than miimon-frequency"
+                    }
+                )
+            }
+            BondMonitoringConfig::ArpMon(config) => {
+                ensure!(
+                    config.interval > 0,
+                    error::InvalidNetConfigSnafu {
+                        reason: "arpmon-interval of 0 disables Arp Monitoring, either set a value or configure Mii Monitoring"
+                    }
+                );
+                // If using Arp Monitoring, 1-16 targets must be specified
+                let targets_length: u32 = config.targets.len() as u32;
+                ensure!(
+                    targets_length > 0 && targets_length <= 16,
+                    error::InvalidNetConfigSnafu {
+                        reason: "arpmon-targets must include between 1 and 16 targets"
+                    }
+                )
+            }
```

Comment:
Since these validations don't depend on any fields from the parent struct, I'd prefer for them to have their own validate methods that get called here.